shared/middleware/helpers/storage_id.rb

```diff
@@ -157,6 +157,7 @@ def storage_id_from_cookie
   return nil if encrypted.empty?
   storage_id = storage_decrypt_id(encrypted)
   return nil if storage_id == 0
+  return nil unless user_storage_ids_table.where(id: storage_id, user_id: nil).first
```

Comment:
FYI @sureshc for HoC scalability that this adds an indexed DB lookup whenever signed-out users access project-backed levels such as the final Dance Party level.